### PR TITLE
Add License dropdown 

### PR DESCRIPTION
**Pull request by binni979**
This is the license field task - create a dropdown with the licenses from [this](https://github.com/spdx/license-list-data/blob/main/json/licenses.json) list.

I added a license dropdown in both workflow and collection.
This is what it looks like after adding a dropdown.
Screenshot 2024-08-20 at 9 42 38 AM

### CODE REVIEW

File: src/pages/Workflows/WorkflowPage.tsx [hunk 1]
```diff
@@ -571,10 +603,17 @@ const WorkflowPage: React.FC = () => {
...
+                placeholder="Select License"
                 {...register("license", { required: "License is required" })}
-              />
+              >
+                {licenses.map((license) => (
+                  <option key={license.licenseId} value={license.licenseId}>
+                    {license.licenseId} - {license.name}
+                  </option>
+                ))}
+                <option value="other">Others</option>
```
Comment (slesaad) [inline src/pages/Workflows/WorkflowPage.tsx]:
```suggestion
                <option value="other">Other</option>
```

File: src/pages/CollectionForm/index.tsx [hunk 2]
```diff
@@ -119,11 +123,28 @@ function CollectionForm() {
...
+              defaultValue=""
+              render={({ field }) => (
+                <Select {...field} id="license" overflowY="auto">
+                  <option value="" disabled>Select a license</option>
+                  {licenses.map((license) => (
+                    <option key={license.licenseId} value={license.licenseId}>
+                      {license.licenseId} - {license.name}
+                    </option>
+                  ))}
+                  <option value="other">Others</option>
```
Comment (slesaad) [inline src/pages/CollectionForm/index.tsx]:
```suggestion
                  <option value="other">Other</option>
```

**Comment by slesaad** [comment]
Added some comments in this issue - https://github.com/NASA-IMPACT/stac-admin/issues/5

File: src/pages/CollectionForm/useUpdateCollection.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { LoadingState, ApiError } from "../../types";
 import { defaultData } from "./constants/updateDataDefaultValue";
 
 type UseUpdateCollectionType = {
-  update: (data: StacCollection, isEditMode: boolean) => Promise<StacCollection>;
+  update: (data: StacCollection, isEditMode: boolean) => Promise<string>;
```
Comment (slesaad) [inline src/pages/CollectionForm/useUpdateCollection.ts]: why switch to `string`?

File: src/services/licenseService.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+export interface License {
+  licenseId: string;
+  name: string;
+}
+
+export const fetchLicenses = async (): Promise<License[]> => {
+  try {
+    const response = await fetch("https://raw.githubusercontent.com/spdx/license-list-data/main/json/licenses.json");
```
Comment (slesaad) [inline src/services/licenseService.ts]: can you move this over to a constants file?

Comment (slesaad) [reply src/services/licenseService.ts]: i just meant the url, not the whole thing

Comment (binni979) [reply src/services/licenseService.ts]: Moved the license url to constant file

**Review by slesaad — APPROVED** [review]

**Pull request merged**